src/main/scala/mesosphere/marathon/core/storage/StorageConfig.scala

```diff
@@ -29,10 +29,11 @@ import scala.collection.immutable.Seq
 import scala.concurrent.ExecutionContext
 import scala.concurrent.duration.{ Duration, _ }
 import scala.reflect.ClassTag
+// scalastyle:on
 
 sealed trait StorageConfig
```

Comment:
To 1) I am 100% with you, that using scallop directly is a bad idea (at least it would be consistent). Let's discuss f2f how we want to proceed with configuration. 

To 2) We want to create a library, which can be reused elsewhere (e.g. metronome). The `store` module is general purpose and can be reused  (and could be part of this library), but the `storage` module is Marathon specific. That's why it would make sense to have 2 modules (perhaps rename `storage` into `repository`)